### ISSUE TITLE
EntityQuery accessCheck() false positive if chaining conditions

### ISSUE THREAD

**Issue by pfrenssen** [mglaman/phpstan-drupal#613]
# Bug report

When an `EntityQuery::accessCheck()` call is chained with other methods it is not detected.

This is similar to #437 but possibly requires a different fix.

### Code snippet that reproduces the problem

This yields a `Relying on entity queries to check access by default is deprecated` error:

```
  $storage = \Drupal::entityTypeManager()->getStorage('node');
  $query = $storage->getQuery();
  $query
    ->accessCheck(FALSE)
    ->condition('field_myfield', TRUE);
  $entity_ids = $query->execute();
```

The ordering doesn't matter, if the `accessCheck()` is last, the error still occurs:

```
  $storage = \Drupal::entityTypeManager()->getStorage('node');
  $query = $storage->getQuery();
  $query
    ->condition('field_myfield', TRUE)
    ->accessCheck(FALSE);
  $entity_ids = $query->execute();
```

When the calls are not chained, the rule works correctly. This does not yield an error:

```
  $storage = \Drupal::entityTypeManager()->getStorage('node');
  $query = $storage->getQuery();
  $query->accessCheck(FALSE);
  $query->condition('field_myfield', TRUE);
  $entity_ids = $query->execute();
```

**Comment by mglaman** [comment]
Thanks for the reproducible code. I'm guessing it's due to an un-chained `execute` because the internal type gets lost after `condition`. The `$query` variable "resets" back to the normal PHP type.

**Comment by sonnykt** [comment]
I notice that chaining is okay as long as `accessCheck()` is followed by `execute()`:

**Comment by joachim-n** [comment]
I am getting a false positive with both versions of this code:

```
    $query
      ->accessCheck(FALSE)
      ->condition($query->andConditionGroup()
        // The interplay between the 'status' and 'workflow' fields is not
        // clearly defined, so we expect both to be in a valid state.
        // See https://factorial-io.atlassian.net/browse/WSD-1965.
        ->condition('status', TRUE)
        ->condition('workflow', ProjectModeration::PUBLISHED)
      )
      ->sort('last_retested')
      ->pager($pager);

    $project_ids = $query->execute();
```

and:

```
    $query
      ->condition($query->andConditionGroup()
        // The interplay between the 'status' and 'workflow' fields is not
        // clearly defined, so we expect both to be in a valid state.
        // See https://factorial-io.atlassian.net/browse/WSD-1965.
        ->condition('status', TRUE)
        ->condition('workflow', ProjectModeration::PUBLISHED)
      )
      ->sort('last_retested')
      ->pager($pager)
      ->accessCheck(FALSE);

    $project_ids = $query->execute();
```

**Comment by claudiu-cristea** [comment]
With version 2.0.13, this is also a false positive

```php
$query = \Drupal::entityTypeManager()->getStorage('tmgmt_job')->getQuery();
      $query->accessCheck(FALSE)
        ->condition('source_language', $row['source language'])
        ->condition('target_language', $row['target language'])
        ->condition('translator', $row['plugin']);
      $ids = $query->execute();
```
